clarifai/client/user.py

```diff
@@ -0,0 +1,48 @@
+from clarifai_grpc.grpc.api import resources_pb2, service_pb2  # noqa: F401
+
+from clarifai.client.app import App
+from clarifai.client.base_auth import BaseAuth
+
+
+class User(BaseAuth):
+  """
+  User is a class that provides access to Clarifai API endpoints related to user information.
+  Inherits from BaseAuth for authentication purposes.
+  """
+
+  def __init__(self, user_id: str, **kwargs):
+    """
+    Initializes a User object.
+    Args:
+        user_id (str): The user ID for the user to interact with.
+        **kwargs: Additional keyword arguments to be passed to the ClarifaiAuthHelper.
+    """
+    self.kwargs = {**kwargs, 'id': user_id}
+    self.user_info = resources_pb2.User(**self.kwargs)
+    super().__init__(user_id=self.id)
```

Comment:
`self.id` isn't defined anywhere.

I also see `self.id` called in other places.
`id` is quite generic a variable name for use in this case since dataset_id, user_id, app_id, etc can all be ids

Reply:
No, The respective protos expects as just `id`
https://github.com/Clarifai/clarifai/blob/8b1af67bbe3d83f06aa799a3b1992a9883b32189/proto/clarifai/api/resources.proto#L2571.

For `App` class `id` refers to app_id and we use `user_id` 
https://github.com/Clarifai/clarifai/blob/8b1af67bbe3d83f06aa799a3b1992a9883b32189/proto/clarifai/api/resources.proto#L75

Reply:
right, but self.id should still throw an error given it has no definition

Reply:
We use [getattr](https://github.com/Clarifai/clarifai-python/blob/96b59b56a8e28b24f6a587f3d192a3c06b040120/clarifai/client/user.py#L39) function to get all user_info params

Reply:
Cool, hadn't seen that method.